modules/apps/marketplace/marketplace-js-components-web/test/js/core/MarketplaceRest.spec.ts

```diff
@@ -0,0 +1,359 @@
+	Cart,
+	MarketplaceConfiguration,
+} from '../../../src/main/resources/META-INF/resources/js/types';
+import product from '../__mock__/product';
+
+const baseResourceURL = 'http://localhost:8080';
+
+const marketplaceConfiguration: MarketplaceConfiguration = {
+	serviceURL: 'https://backend.marketplace.liferay.com',
+	settings: {
+		account: {id: 123, name: 'Liferay Labs'},
+		channelId: 123,
+		cloudProject: 'exte5a2marketplace-extuat',
+		references: {fragmentsFilter: '', paymentMethodFilter: ''},
+		siteId: 123,
+		userAccount: {id: 123, name: 'Ray'},
+	},
+	url: 'https://marketplace.liferay.com',
+};
+
+const marketplaceRest = new MarketplaceRest(
+	baseResourceURL,
+	marketplaceConfiguration
+);
+
+const marketplaceToken = {
+	accessToken: 'abcdef',
+	accessTokenExpirationTime: (new Date().getTime() + 60000).toString(),
+};
+
+jest.mock('frontend-js-web', () => {
+	const actual = jest.requireActual('frontend-js-web');
+
+	return {
+		...actual,
+		createResourceURL: jest.fn((baseURL, params) => ({
+			href: `${baseURL}?${new URLSearchParams(params).toString()}`,
+			toString: () =>
+				`${baseURL}?${new URLSearchParams(params).toString()}`,
+		})),
+		fetch: mockFetch,
+		openToast: jest.fn(),
+		sub: jest.fn((langKey, arg) => langKey.replace('x', arg)),
+	};
+});
+
+describe('MarketplaceRest', () => {
+	let getItemFn: jest.Mock<WindowSessionStorage['sessionStorage']['getItem']>;
+	let setItemFn: jest.Mock<WindowSessionStorage['sessionStorage']['setItem']>;
+
+	beforeEach(() => {
+		getItemFn = jest
+			.fn()
+			.mockImplementation(() => JSON.stringify(marketplaceToken));
+		setItemFn = jest.fn();
+
+		globalThis.Liferay.Util.SessionStorage = {
+			TYPES: {
+				NECESSARY: 'Necessary',
+			},
+			getItem: getItemFn,
+			setItem: setItemFn,
+		};
+	});
+
+	it('fetch marketplace token without cache', async () => {
+		const {fetch} = require('frontend-js-web');
+
+		fetch.mockResponseOnce(JSON.stringify(marketplaceToken));
+
+		getItemFn = jest.fn().mockImplementation(() => null);
+
+		globalThis.Liferay.Util.SessionStorage.getItem = getItemFn;
+
+		const marketplaceAuthorization =
+			await marketplaceRest.getMarketplaceToken();
+
+		expect(marketplaceAuthorization).toMatchObject(marketplaceToken);
+		expect(getItemFn).toBeCalledTimes(1);
+		expect(setItemFn).toBeCalledTimes(1);
+
+		const [firstSetItemCall] = setItemFn.mock.calls;
+
+		expect(fetch).toBeCalledTimes(1);
+		expect(firstSetItemCall[0]).toBe('@marketplace/token');
+		expect(firstSetItemCall[1]).toBe(JSON.stringify(marketplaceToken));
+		expect(firstSetItemCall[2]).toBe(
+			globalThis.Liferay.Util.SessionStorage.TYPES.NECESSARY
+		);
+	});
+
+	it('fetch marketplace token from cache', async () => {
+		const marketplaceAuthorization =
+			await marketplaceRest.getMarketplaceToken();
+
+		expect(fetch).toBeCalledTimes(0);
+		expect(getItemFn).toBeCalledTimes(1);
+		expect(setItemFn).toBeCalledTimes(0);
+		expect(marketplaceAuthorization).toMatchObject(marketplaceToken);
```

Comment:
here we have 2 different patterns for the same test
Take a look at the expect() sort